### ISSUE TITLE
Load only encoder part of model

### ISSUE THREAD

**Issue by starsong98** [facebookresearch/segment-anything#521]
Hello, thank you for releasing this model's code as well as pretrained weights.

How can I load only the image encoder (ViT) portion of the pretrained SAM model? I want to use it as a backbone for a different downstream task.

**Comment by heyoeyo** [comment]
The image encoder model can be found the in [image_encoder.py](https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/modeling/image_encoder.py) file, specifically it's the `ImageEncoderViT` class that you'd want to use.

You'd have to instantiate that class with the appropriate input arguments (which depend on the size of the SAM model you want to use, i.e. vit-b vs vit-l vs vit-h), you can see what those arguments are from the [build_sam.py](https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/build_sam.py) file.

Then finally, you'd load the appropriate state dictionary (i.e. checkpoint file). However, the checkpoints assume that you're using a model (class) that has the image encoder stored under a variable called image_encoder (i.e. `self.image_encoder`), since that's how the [SAM model is structured](https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/segment_anything/modeling/sam.py#L43). 
You'd need to match this structure if you just wanted to be able load the state dictionary as-is (you'd also need to make sure to not use `strict` mode when using [load_state_dict](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.load_state_dict), since the checkpoints contain the prompt/mask model weights, that likely won't match your custom model). Alternatively, if you're comfortable editing the state dict, you could just modify it to match whatever your model expects.

**Comment by starsong98** [comment]
Thank you! Setting the strict=False was the key to making it work.
